src/roman_datamodels/datamodels/_core.py

```diff
@@ -38,7 +38,10 @@ def wrapper(self, *args, **kwargs):
         if self._asdf is None:
             try:
                 with validate.nuke_validation():
-                    self._asdf = asdf.AsdfFile({"roman": self._instance})
+                    af = asdf.AsdfFile()
+                    af["roman"] = self._instance
+                    af.validate()
```

Comment:
This seems a bit verbose. Is there something wrong with:
```python
self._asdf = asdf.AsdfFile({"roman": self._instance})
self._asdf.validate()
```

Reply:
Thanks for taking a look and good question. Unfortunately, the code you proposed would validate the instance twice. Once during `AsdfFile.__init__` and again during `AsdfFile.validate`. Also, the reason for this PR is that asdf is deprecating validation during `AsdfFile.__init__`. The code you proposed would result in a `DeprecationWarning` and test failure when the validation fails. The code proposed in this PR avoids the possibility of the `DeprecationWarning` by not providing the `_instance` during `AsdfFile.__init__.`.

Reply:
Is asdf deprecating passing a dictionary directly to the `AsdfFile` initializer? If so, then that is a huge change of behavior!

If it's simply raising `DepreciationWarning` because of the validation change then a more thoughtful change should be considered as initializing with a dictionary is an extremely common usage of `AsdfFile`.

I would request that a `validate=True` option should be added to the initializer which preserves the current behavior without deprecation. You can set it so it defaults to `None` and raises the warning of impending behavior changes in that case.

Reply:
asdf is not deprecating initializing with a dictionary. However if that dictionary is invalid one will see a `DeprecationWarning` noting that validation on `AsdfFile.__init__` is deprecated and a `ValidationError`. If the dictionary is valid no warning or error will be seen. This was done to allow test suites (like the one in `roman_datamodels` to detect the change (the warning appears during the CI run which gets turned into an error).

Reply:
@WilliamJamieson I'm not sure I addressed your comments here. asdf is deprecating (for removal) validation on `AsdfFile.__init__` to reduce the duplication of functionality in asdf. Adding a `validate` keyword that triggers validation overlaps with functionality that is already provided by `AsdfFile.validate`.